### ISSUE TITLE
Document type classes in @polkadot/types

### ISSUE THREAD

**Issue by jacogr** [polkadot-js/api#376]
We need to start linking through to these, the first order of business is to document the basics in them -

- name, description
- getters
- static methods should be explicitly excluded from the docs (i.e. `U8a.decodeU8a` should not be included, since these are meant to be only used internally, neither any other statics in the classes). There is one exception to this blanket ban on static docs - the `Compact` methods to manage (read, write, add) lengths are utility functions and should be in.
- Do not document (exclude explicitly) functions that should be replaced by extending - specifically here we have the `add`, `sub`, `eq`, `isZero` on the `UInt` classes - since that will extend `BN` (soon-ish), don't put any focus on them.

In short -

- docs should be useful
- it should be linkable from other places, e.g. doing an in-doc `[[ MyFantasticType ]]` should link through to the `MyFantasticType` class (this way we can use it from e.g. the api classes)
- it should focus on the descriptions and getters
- it should not have internal-only methods not fields (no fields at all)
